components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorConsoleService.java

```diff
@@ -0,0 +1,144 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.siddhi.editor.core.internal;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.log4j.helpers.LogLog;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.editor.log.appender.internal.CircularBuffer;
+import org.wso2.carbon.editor.log.appender.internal.ConsoleLogEvent;
+import org.wso2.carbon.editor.log.appender.DataHodlder;
+import org.wso2.msf4j.websocket.WebSocketEndpoint;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.websocket.OnClose;
+import javax.websocket.OnMessage;
+import javax.websocket.OnOpen;
+import javax.websocket.Session;
+import javax.websocket.server.ServerEndpoint;
+
+/**
+ * EditorConsoleService Websocket - Client connect to this
+ */
+@Component(
+        name = "editor-console-service",
+        service = WebSocketEndpoint.class,
+        immediate = true
+)
+@ServerEndpoint(value = "/console")
+public class EditorConsoleService implements WebSocketEndpoint {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EditorConsoleService.class);
+    private static Queue<Session> queue = new ConcurrentLinkedQueue<Session>();
+    private CircularBuffer<ConsoleLogEvent> circularBuffer = DataHodlder.getBuffer();
+    private ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(10);
+    private int schedulerInitialDelay = 1000;
+    private int schedulerTerminationDelay = 10;
+    private int publishedIndex = -1;
+
+
+    @OnOpen
+    public void onOpen(Session session) {
+        LOGGER.info("Connected with user : " + session.getId());
+        queue.add(session);
+        scheduler.scheduleWithFixedDelay(new LogPublisherTask(), schedulerInitialDelay, schedulerTerminationDelay,
+                TimeUnit.MILLISECONDS);
+    }
+
+    @OnMessage
+    public void onMessage(String text, Session session) {
+        try {
+            session.getBasicRemote().sendText("Welcome to Stream Processor Studio");
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage());
+        }
+    }
+
+    @OnClose
+    public void onClose(Session session) {
+        if (session.isOpen()) {
+            try {
+                session.close();
+            } catch (IOException e) {
+                LOGGER.error(e.getMessage());
```

Comment:
Have more context on all the error messages and have the e parsed too

Reply:
Fixed with commit [#11d18ac2bd9d5463a9b2562066d005e5472a7ca0](https://github.com/wso2/carbon-analytics/commit/11d18ac2bd9d5463a9b2562066d005e5472a7ca0)